apiserver/facades/client/charms/client.go

```diff
@@ -603,19 +603,19 @@ func validateOrigin(origin params.CharmOrigin, schema string, switchCharm bool)
 }
 
 func (a *API) getCharmRepository(src corecharm.Source) (corecharm.Repository, error) {
-	a.mu.Lock()
```

Comment:
Why is the lock being removed?

Reply:
As we only trigger the nil check for the cached repoFactory with tests, it's not actually required anymore.